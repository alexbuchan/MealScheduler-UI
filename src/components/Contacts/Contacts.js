import React from 'react';
import PropTypes from 'prop-types';
import ContactCard from './ContactCard/ContactCard';

const propTypes = {
  contacts: PropTypes.array
};

const Contacts = ({ contacts }) => {
  if (!(contacts.length === 0)) {
    return contacts.map((contact) => {
      return <ContactCard key={ contact.email } contact={ contact } />
    });
  }

  return null;
}

Contacts.propTypes = propTypes;
export default Contacts;