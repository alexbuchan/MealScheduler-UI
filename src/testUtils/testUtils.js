import React from 'react';
import { shallow } from 'enzyme';

export const componentSetup = (Component, props = {}, options = {}) => {
  const instance = shallow(<Component { ...props } />, options);
  return instance;
}

export const findByTestAttribute = (instance, attribute) => {
  const wrapper = instance.find(`[data-test="${attribute}"]`);
  return wrapper;
}