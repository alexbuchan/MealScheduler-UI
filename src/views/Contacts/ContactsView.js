import React from 'react';
import { Redirect, Link } from 'react-router-dom';
import ContactActions from '../../actions/contact/ContactActions';
import ContactStore from '../../stores/ContactStore/ContactStore';
import UserStore from '../../stores/UserStore/UserStore';
import Background from '../../components/Background/Background';

class ContactsView extends React.Component {
  constructor() {
    super();

    this.state = {
      auth: UserStore.getUserState().auth,
      contacts: ContactStore.getContactState().contacts
    }

    this._onChange = this._onChange.bind(this);
  }

  _onChange() {
    this.setState({
      userStore: UserStore.getUserState(),
      contacts: ContactStore.getContactState().contacts
    });
  }

  componentDidMount() {
    ContactActions.getContacts();
    UserStore.addChangeListener(this._onChange);
    ContactStore.addChangeListener(this._onChange)
  }

  componentWillUnmount() {
    UserStore.removeChangeListener(this._onChange);
    ContactStore.removeChangeListener(this._onChange);
  }

  renderContactAddress = (contact) => {
    return <ul className="address-list">{ Object.entries(contact.address).map((item, index) => {
      if (item[0] === 'geo') return null;
      const label = item[0].replace(/^\w/, c => c.toUpperCase());
      return (
        <li className="contact-item address-item" key={ index }><strong>{ label }</strong>: { item[1] }</li>
      );
    }) }</ul>
  }

  renderContacts = () => {
    if (!(this.state.contacts.length === 0)) {
      return this.state.contacts.map((contact) => {
        return (
          <ul key={ contact.email } className="contact-info">
            <div className="contact-name-wrapper">
              <h3 className="contact-name">{ contact.name }</h3>
            </div>
            <div className="contact-info-list">
              <li className="contact-item"><strong>Username</strong>: { contact.username }</li>
              <li className="contact-item"><strong>Email</strong>: { contact.email }</li>
              <li className="contact-item"><strong>Address</strong>: { this.renderContactAddress(contact) }</li>
              <li className="contact-item"><strong>Phone</strong>: { contact.phone }</li>
              <li className="contact-item"><strong>Website</strong>: { contact.website }</li>
            </div>
          </ul>
        );
      });
    }

    return null;
  }

  render() {
    if (!this.state.auth) {
      return <Redirect to='/login' />;
    }

    return (
      <div className="contacts-view">
        <Background />
        <h1 className="contacts-title">Contacts</h1>
        <div className="contacts-info-wrapper">
          { this.renderContacts() }
        </div>
      </div>
    );
  }
}

export default ContactsView;