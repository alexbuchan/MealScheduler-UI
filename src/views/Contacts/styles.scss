.contacts-view {
  height: $contentHeight;
  display: flex;
  flex-direction: column;
  align-items: center;

  .contacts-title {
    margin: 30px;
    font-size: xx-large;
  }

  .contacts-info-wrapper {
    display: flex;
    justify-content: center;
    box-shadow: 0 0 10px;
    width: 80%;
    height: 100%;
    margin-bottom: 10px;
    flex-wrap: wrap;
    overflow: scroll;
    background-color: rgba(255, 255, 255, 0.7);

    .contact-info {
      border: 1px solid gray;
      box-shadow: 0 0 8px;
      padding: 0;
      margin: 10px;
      width: 250px;
      height: 315px;
      background-color: ghostwhite;

      .contact-name-wrapper {
        display: flex;
        justify-content: center;
        align-items: center;
        height: 75px;
        box-shadow: 0 5px 5px grey;
        background-color: bisque;

        .contact-name {
          text-align: center;
          font-size: x-large;
          width: 88%;
        }
      }

      .contact-info-list {
        margin: 10px;
        list-style: none;

        .contact-item {
          font-size: 14px;
          margin: 2px;
           
          .address-list {
            margin-bottom: 12px;
          }

          .address-item {
            list-style: none;
          }
        }
      }
    }
  }

  .contacts-info-wrapper::-webkit-scrollbar {
    display: none;
  }
  
  .contacts-info-wrapper {
    -ms-overflow-style: none;
  }
}