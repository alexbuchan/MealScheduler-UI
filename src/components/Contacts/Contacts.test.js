import Contacts from './Contacts';
import { componentSetup } from '../../testUtils/testUtils';

describe('Contacts', () => {
  describe('when contacts as props are NOT passed', () => {
    let instance;
    let props;
    
    beforeEach(() => {
      props = {
        contacts: []
      };

      instance = componentSetup(Contacts, props);
    });

    it('should return null', () => {
      expect(instance.type()).toBeNull();
    });
  });

  describe('when contacts as props are passed', () => {
    let instance;
    let props;
    
    beforeEach(() => {
      props = {
        contacts: [
          { name: 'Contact1', email: 'contact1@email.com' },
          { name: 'Contact2', email: 'contact2@email.com' }
        ]
      };
  
      instance = componentSetup(Contacts, props);
    });

    it('Should render without errors', () => {
      expect(instance).toMatchSnapshot();
    });
  });
});