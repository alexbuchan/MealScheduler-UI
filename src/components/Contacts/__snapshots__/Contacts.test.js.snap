// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts when contacts as props are passed Should render without errors 1`] = `
Array [
  <ContactCard
    contact={
      Object {
        "email": "contact1@email.com",
        "name": "Contact1",
      }
    }
    key="contact1@email.com"
  />,
  <ContactCard
    contact={
      Object {
        "email": "contact2@email.com",
        "name": "Contact2",
      }
    }
    key="contact2@email.com"
  />,
]
`;
